pkg/controllers/overrider/common_test.go

```diff
@@ -280,3 +216,197 @@ var _ = Describe("Test ClusterResourceOverride common logic", func() {
 		})
 	})
 })
+
+var _ = Describe("Test ResourceOverride common logic", func() {
+	var ro *fleetv1alpha1.ResourceOverride
+	totalSnapshots := 7
+	testROName := "test-ro-common"
+	var namespaceName string
+
+	BeforeEach(func() {
+		namespaceName = fmt.Sprintf("%s-%s", overrideNamespace, utils.RandStr())
+		namespace := &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		}
+		Expect(k8sClient.Create(ctx, namespace)).Should(Succeed())
+		// we cannot apply the RO to the cluster as it will trigger the real reconcile loop.
+		ro = getResourceOverride(testROName, namespaceName)
+		By("Creating resourceOverrideSnapshot")
+		for i := 0; i < totalSnapshots; i++ {
+			snapshot := getResourceOverrideSnapshot(testROName, namespaceName, i)
+			Expect(k8sClient.Create(ctx, snapshot)).Should(Succeed())
+		}
+	})
+
+	AfterEach(func() {
+		By("Deleting five clusterResourceOverrideSnapshot")
+		for i := 0; i < totalSnapshots; i++ {
+			snapshot := getResourceOverrideSnapshot(testROName, namespaceName, i)
+			Expect(k8sClient.Delete(ctx, snapshot)).Should(SatisfyAny(Succeed(), &utils.NotFoundMatcher{}))
+		}
+		By("Deleting the namespace")
+		namespace := &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		}
+		Expect(k8sClient.Delete(ctx, namespace)).Should(SatisfyAny(Succeed(), &utils.NotFoundMatcher{}))
```

Comment:
i find the utils.NotFoundMatcher can be replaced by "WithTransform(errors.IsNotFound, BeTrue())" when working on the atm features

Reply:
good to know but I think we can keep this matcher in this repo